### PR TITLE
chore: improve contributing guidelines

### PR DESCRIPTION
**Pull request by fmvilas**
**Description**

- Adds information about the new stages: strawman, proposal, draft, and accepted (or rejected).
- Adds some guiding principles.

**To-do**

- [ ] Clean labels of this repository and add a few ones for each stage, `Needs Champion`, etc. Look at https://github.com/graphql/graphql-spec/labels for inspiration.

**Related issue(s)**
https://github.com/asyncapi/.github/pull/37

### CODE REVIEW

File: CONTRIBUTING.md
Comment (magicmatatjahu) [inline CONTRIBUTING.md:193]: Do you mean by `AsyncAPI.js` AsyncAPI JS Parser?

File: CONTRIBUTING.md
Comment (magicmatatjahu) [inline CONTRIBUTING.md:96]: Hah 🤣

**Review by magicmatatjahu — COMMENTED** [review]
One comment :) I think that the text is correct - because is copied entirely from graphql 😂

File: CONTRIBUTING.md
Comment (fmvilas) [reply CONTRIBUTING.md:193]: Hahahaha! Good catch.

File: CONTRIBUTING.md
Comment (fmvilas) [reply CONTRIBUTING.md:96]: 😄 No shame. That's open source. They did a great job and so we reuse it.

**Review by magicmatatjahu — APPROVED** [review]
🆗 Please wait for rest of mates :)

**Comment by derberg** [comment]
who is the champion? I mean who can be a champion? where can I find a champion for my case?

**Comment by smoya** [comment]
> who is the champion? I mean who can be a champion? where can I find a champion for my case?

+1 to this question.

File: CONTRIBUTING.md
Comment (jonaslagoni) [inline CONTRIBUTING.md:163]: This does not sound quite right to me 🤔 You mean the problem has been identified and agreed upon or?

File: CONTRIBUTING.md
Comment (jonaslagoni) [inline CONTRIBUTING.md:199]: > fully tested and merged or ready to merge

🤔 Do you mean merged into the RFC branch and ready to be merged into `master` or? 🤔

**Review by jonaslagoni — COMMENTED** [review]
Does this mean that a `champion` MUST know how to code JavaScript? 🤔 Or can they have something like `co-champions`?

**Comment by fmvilas** [comment]
> who is the champion? I mean who can be a champion?

A champion is anyone who wants to take the lead on implementing the feature. It doesn't mean it's the only person working on it though. Think about this role as a "person of reference".

> where can I find a champion for my case?

Usually in the issue itself. Anyone can jump in and say they want to champion that feature. Or in case the old champion isn't responding and there's a PR already, a new person can comment on the PR.

> Does this mean that a champion MUST know how to code JavaScript? 🤔 Or can they have something like co-champions?

A champion is a "person of reference" for that feature. The champion may or may not know about JS and rely on the help of other community members to implement the feature on the JS parser.

File: CONTRIBUTING.md
Comment (fmvilas) [reply CONTRIBUTING.md:163]: I'm not sure what is that you don't understand there 😅

File: CONTRIBUTING.md
Comment (fmvilas) [reply CONTRIBUTING.md:199]: Merged to `master`. The RFC branch is where you probably work on the feature and once it's done, it's merged to `master` or whatever is the main branch in that repo (now and in the future).

**Comment by derberg** [comment]
@fmvilas well explained, will you add it? 😄

**Comment by fmvilas** [comment]
Ok! I thought it was a common term, like "maintainers". I'll add it anyway.

File: CONTRIBUTING.md
Comment (jonaslagoni) [reply CONTRIBUTING.md:199]: Gotcha 👍

File: CONTRIBUTING.md
Comment (jonaslagoni) [reply CONTRIBUTING.md:163]: never mind, read it a couple of times and it makes sense now 🤷

**Comment by jonaslagoni** [comment]
> Ok! I thought it was a common term, like "maintainers". I'll add it anyway.

Would probably be great to know  

> A champion is a "person of reference" for that feature. The champion may or may not know about JS and rely on the help of other community members to implement the feature on the JS parser.

Would probably be great to add the part about they are not expected to know JS and do the implementation on their own. Just to ensure people arent "scared" away by the fact that there has to be some underlying code implementation.

**Comment by fmvilas** [comment]
Added the definition of "champion" and linked it in the first occurrence on the document.

**Review by derberg — APPROVED** [review]
I think going GraphQL way, meaning already battle-proof way is the best we can do here 👍🏼 
I approve but suggest leaving it open for few days as maybe someone from the community has some comments

**Review by jonaslagoni — APPROVED** [review]
👍

**Comment by fmvilas** [comment]
@derberg Yes. Would you mind sharing it on social media? I think it deserves some attention.

**Review by magicmatatjahu — APPROVED** [review]

**Review by smoya — DISMISSED** [review]
🚀

**Comment by derberg** [comment]
@fmvilas 
- https://twitter.com/AsyncAPISpec/status/1371446583221452807
- https://www.linkedin.com/feed/update/urn:li:activity:6777212281198563328

**Comment by derberg** [comment]
It is actually not included in the release. This was an unintentional release caused by the release automation initial run. 1.0 version of the spec was released a long time ago. The release mentioned by the bot was removed.

**Comment by derberg** [comment]
Not included in release 2.1 yet, 2.1 was mistakenly triggered by our automation.

**Pull request merged**
